### DISCUSSION TITLE
Exporting to *.rpmap / Maptool

### DISCUSSION THREAD

**Discussion by CelesteEvrad** [mapeditor/tiled#3184] [Q&A]
When I import the .rpmap file the tile size is wrong they are showing up as 300x300 and massively overlapping each other. The tiles are only supposed to be 16x16. Any Ideas on what I am doing wrong?

**Comment by bjorn** [comment]
Hmm, can you share the map you're exporting?

Maybe @cpetig could help since I have personally no experience with MapTool.

**Reply by bjorn** [reply]
@CelesteEvrad Hmm, please attach the file instead, this is too much text. Also, I think the Tiled map file that you're exporting would be more helpful.

**Reply by CelesteEvrad** [reply]
Does this help?

[TestTiled.zip](https://github.com/mapeditor/tiled/files/7609323/TestTiled.zip)
[TestTiled.rpmap.zip](https://github.com/mapeditor/tiled/files/7609324/TestTiled.rpmap.zip)

**Reply by cpetig** [reply]
Thanks for uploading, I will take a look (but don't promise it will be this week)

**Reply by cpetig** [reply]
Reading your original problem description I believe the error is me hardcoding the size in the exporter - having your testcase is a good motivation to look into it.

**Reply by CelesteEvrad** [reply]
^_^ Thank you for looking into it

**Reply by bjorn** [reply]
Thanks @cpetig! I've pushed a quick fix for the hardcoded 300x300 tile size (d952a1ecb03059b5fcd5dd379157a9b1d86cbeef), which should address this problem. Builds with this fix should soon be available at https://github.com/mapeditor/tiled/actions/runs/1525261468 for testing.

**Reply by CelesteEvrad** [reply]
^_^

**Reply by CelesteEvrad** [reply]
Hey thanks for the quick turnaround. Just one more thing.
When exporting it only exports tile layer 1. I wanted to know if this was a limitation or if this was a feature that hadn't been implemented yet? It would be really useful to export multiple layers (buildings on top of grass for example)

**Reply by bjorn** [reply]
Exporting multiple tile layers is not implemented yet, but I'm not sure what MapTool expects in this case in terms of tile rendering order. Maybe it's something @cpetig would still like to add or just shed some light on, so I could have a go at it.
